extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/HttpAuthenticator.java

```diff
@@ -377,6 +400,42 @@ public void accept(HttpCredentialTransport t) {
                 });
     }
 
+    private static void addBasicAuthMechanismIfImplicitlyRequired(
+            Instance<HttpAuthenticationMechanism> httpAuthenticationMechanism,
+            List<HttpAuthenticationMechanism> mechanisms, Instance<IdentityProvider<?>> providers) {
+        if (!Boolean.getBoolean(TEST_IF_BASIC_AUTH_IMPLICITLY_REQUIRED) || isBasicAuthNotRequired()) {
+            return;
+        }
+
+        var basicAuthMechInstance = httpAuthenticationMechanism.select(BasicAuthenticationMechanism.class);
+        if (basicAuthMechInstance.isResolvable() && !mechanisms.contains(basicAuthMechInstance.get())) {
```

Comment:
This is a useful check, but I believe it should be running all the time, it will save time to users who forgot to add the right identity provider's configuration

Reply:
here is BasicAuthentication or auth-mechanism=basic I think that's what you mean. we don't do that for other mechanisms either. some validation would be nice, sure. IMO not at this place of the `Httpauthenticator` because that needs to be done during the app startup.

Reply:
I'll add some additional validation that will inform user.

Reply:
Hmm, thinking about it. I think this debug message does it for cases where it is safe to done. We can't really be sure to log anything if we don't know that basic auth is required and there is no other mechanism that provides it.